tasks/mpi/filateva_e_metod_gausa/func_tests/main.cpp

```diff
@@ -0,0 +1,281 @@
+  std::vector<double> answer;
+  std::vector<double> tResh;
+
+  std::shared_ptr<ppc::core::TaskData> taskData = std::make_shared<ppc::core::TaskData>();
+
+  if (world.rank() == 0) {
+    matrix.resize(size * size);
+    vecB.resize(size);
+    tResh = gereratorSLU(matrix, vecB);
+
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(matrix.data()));
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(vecB.data()));
+    taskData->inputs_count.emplace_back(size);
+    taskData->outputs_count.emplace_back(size);
+  }
+
+  filateva_e_metod_gausa_mpi::MetodGausa metodGausa(taskData);
+
+  ASSERT_TRUE(metodGausa.validation());
+  metodGausa.pre_processing();
+  metodGausa.run();
+  metodGausa.post_processing();
+
+  if (world.rank() == 0) {
+    auto *temp = reinterpret_cast<double *>(taskData->outputs[0]);
+    answer.insert(answer.end(), temp, temp + size);
+
+    EXPECT_EQ(answer.size(), tResh.size());
+    for (int i = 0; i < size; i++) {
+      EXPECT_NEAR(tResh[i], answer[i], alfa);
+    }
+  }
+}
+
+TEST(filateva_e_metod_gausa_mpi, test_size_200) {
+  boost::mpi::communicator world;
+  int size = 200;
+  std::vector<double> matrix;
+  std::vector<double> vecB;
+  std::vector<double> answer;
+  std::vector<double> tResh;
+
+  std::shared_ptr<ppc::core::TaskData> taskData = std::make_shared<ppc::core::TaskData>();
+
+  if (world.rank() == 0) {
+    matrix.resize(size * size);
+    vecB.resize(size);
+    tResh = gereratorSLU(matrix, vecB);
+
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(matrix.data()));
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(vecB.data()));
+    taskData->inputs_count.emplace_back(size);
+    taskData->outputs_count.emplace_back(size);
+  }
+
+  filateva_e_metod_gausa_mpi::MetodGausa metodGausa(taskData);
+
+  ASSERT_TRUE(metodGausa.validation());
+  metodGausa.pre_processing();
+  metodGausa.run();
+  metodGausa.post_processing();
+
+  if (world.rank() == 0) {
+    auto *temp = reinterpret_cast<double *>(taskData->outputs[0]);
+    answer.insert(answer.end(), temp, temp + size);
+
+    EXPECT_EQ(answer.size(), tResh.size());
+    for (int i = 0; i < size; i++) {
+      EXPECT_NEAR(tResh[i], answer[i], alfa);
+    }
+  }
+}
+
+TEST(filateva_e_metod_gausa_mpi, test_size_different) {
+  boost::mpi::communicator world;
+  int size = 5;
+  std::vector<double> matrix;
+  std::vector<double> vecB;
+
+  std::shared_ptr<ppc::core::TaskData> taskData = std::make_shared<ppc::core::TaskData>();
+
+  if (world.rank() == 0) {
+    matrix.resize(size * size);
+    vecB.resize(size);
+
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(matrix.data()));
+    taskData->inputs.emplace_back(reinterpret_cast<uint8_t *>(vecB.data()));
+    taskData->inputs_count.emplace_back(size);
+    taskData->outputs_count.emplace_back(size + 1);
+  }
+
+  filateva_e_metod_gausa_mpi::MetodGausa metodGausa(taskData);
+
+  if (world.rank() == 0) {
+    ASSERT_FALSE(metodGausa.validation());
```

Comment:
Ну вот как говорите не писать везде, тут не написал и тут ASSERT а не EXPECT
```suggestion
    EXPECT_FALSE(metodGausa.validation());
```